src/Service/Requests/ElasticsearchRequest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace SilverStripe\DiscovererElasticEnterprise\Service\Requests;
+
+use Elastic\EnterpriseSearch\AppSearch\Schema\EsSearchParams;
+use Elastic\EnterpriseSearch\Request\Request;
+use SilverStripe\Core\Injector\Injectable;
+
+/**
+ * Request to submit elasticsearch queries via the Enterprise search endpoint
+ *
+ * @see https://www.elastic.co/guide/en/app-search/8.17/elasticsearch-search-api-reference.html
+ */
+class ElasticsearchRequest extends Request
```

Comment:
I hate this about Elasticsearch... but I kinda feel like this class should be called `ElasticsearchSearchRequest`, since this is specifically a request to the `_search` endpoint.

I'm not sure that we will have any other Elasticsearch requests in the future, but the class name atm makes it sounds like it's here to cover all Elasticsearch requests.